nikeee/setup-pandoc
# adding some autoated test to the action

I created few tests for the action so that the macOS error that I'm currently facing don't reappear by accident. 

- create GitHub actions triggered by push/PR to master branch 
- install the checkout action 
- test installation process 
- verify version number 
- use a dummy test text and verify the output
- special case for macOS to check a version < 3.1.1 (when the .pkg file name changed)


Challenges: 
- the macOS build don't work. I already created a patch that may solved it in #5 I think it's safe to ignore it in this PR 
- the tested text in windows don't return the expected result. I'm no pandoc expert, is it normal ? 
- the latest version is currently hard coded I don't want to spend more time on this PR at the moment but maybe open an issue to dynamically wire the latest one. 

Demo: 
I checked it on my side: https://github.com/12rambau/setup-pandoc/actions/runs/6417462490
